♻️ refactor: 공고 참여 스웨거 필수 필드 미적용 해결

Jiwoong0322 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #85 
- 작업 요약: 공고 참여에 extend_schema가 적용되지 않아 스웨거에서 필수 필드를 입력할 수 없는 문제 해결
- 
## 📄 상세 내용
- [ ] applications_views.py에 extend_schema 추가

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Jiwoong0322 committed:
♻️ refactor: 공고 참여 스웨거 필수 필드 미적용 해결

 - 공고 참여에 extend_schema가 적용되지 않아 스웨거에서 필수 필드를 입력할 수 없는 문제 해결

관련 이슈: #85

Jiwoong0322 committed:
♻️ refactor: mypy에러 해결

타입 미지정으로 인한 mypy에러 해결

관련 이슈: #85

Umdoong commented on apps/applications/views/applications_views.py:
```diff
@@ -22,39 +23,56 @@
+# extend_schema에 사용하기 위한 Serializer 클래스 (mypy 오류 수정)
+class _ApplicationCreateResponseSerializer(serializers.Serializer[dict[str, Any]]):
+    application_id = serializers.IntegerField()
+    message = serializers.CharField()
```
지원서 생성할 때 request body에 받을 필드가 이게 전부인가요...?

Jiwoong0322 replied:
해당 필드는 참여 신청 성공시 응답용으로 작성한건데 serializers에 합치는걸 깜빡했습니다. request body에 받을 필드는 이걸로 받아오게 했습니다. request={'application/json': ApplicationCreateSerializer}
수정 후 푸쉬하겠습니다

Jiwoong0322 committed:
♻️ refactor: serializer통일

view에 임시로 작성했던 ApplicationCreateResponseSerializer와 ErrorResponseSerializer를 application_serializers.py에 작성.

관련 이슈: #85

Jiwoong0322 committed:
♻️ refactor: isort 적용

isort . 적용

관련 이슈: #85

Umdoong commented:
@Jiwoong0322 
일단 머지할게요
근데 응답데이터가 어플리케이션의 id와 성공 메세지만 있으면 되는지 프론트랑 논의 해보셨는지 궁금하네요
